.scalafmt.conf

```diff
@@ -1,45 +1,140 @@
-version = "2.0.0-RC8"
-maxColumn = 100
-
+version = "3.7.17"
+maxColumn = 120
 encoding = "UTF-8"
+runner.dialect = scala213
 
-docstrings = JavaDoc
+docstrings {
+  style = Asterisk
+  forceBlankLineBefore = false
+  wrap = false
+  oneline = keep
+}
 
-continuationIndent = 4
+danglingParentheses {
+ctrlSite = false
+defnSite = false
+callSite = false
+}
 
-continuationIndent.callSite = 2
-continuationIndent.defnSite = 4
+align {
+  preset = most
+  allowOverflow = true
+  openParenDefnSite = true
+  openParenCallSite = true
+  openParenCtrlSite = true
+  multiline = false
+  inInterpolation = true
+  tokens."+" = [{
+    code = ":"
+    owners = [{
+      parents = ["Defn\\."]
+    },
+      {
+        regex = "Term\\.Param"
+        parents = ["Ctor\\.Primary"]
+      }]
+  }, {
+    code = "=>"
+    owners = [{
+      regex = "Case"
+    }]
+  }
+  ]
+}
 
-align = more
-align.openParenCallSite = false
-align.openParenDefnSite = false
-align.tokens = [
-  {code = "=>", owner = "Case"},
-  {code = "%", owner = "Term.ApplyInfix"},
-  {code = "%%", owner = "Term.ApplyInfix"}  
-]
+newlines {
+  source = keep
+  alwaysBeforeElseAfterCurlyIf = false
+  beforeMultiline = fold
+  afterCurlyLambdaParams = squash
+  avoidForSimpleOverflow = [slc, tooLong, punct] # singe line comment, does not add newline if line would be too long after,  punctuation
+  forceBeforeMultilineAssign = never
+}
 
-newlines.alwaysBeforeTopLevelStatements = false
-newlines.alwaysBeforeElseAfterCurlyIf = false
-newlines.afterCurlyLambda = never
+verticalMultiline {
+  atDefnSite = true
+  arityThreshold = 6
+}
 
-rewrite.rules = [ExpandImportSelectors]
+# TODO Comments newline
+# alignment constructor
 
-rewrite.rules = [RedundantParens]
 
-rewrite.rules = [PreferCurlyFors]
+indent {
+  main = 4
+  callSite = 4
+  ctorSite = 8 # Double Indent
+  withSiteRelativeToExtends = 4
+}
+
+assumeStandardLibraryStripMargin = true
+lineEndings = unix
+binPack.parentConstructors = keep
+includeNoParensInSelectChains = false
+optIn.breakChainOnFirstMethodDot = false
+optIn.configStyleArguments = true
+optIn.breaksInsideChains = false
+rewrite {
+  rules = [
+    RedundantParens,
+    RedundantBraces,
+    SortModifiers,
+    PreferCurlyFors,
+    Imports
+  ]
+  sortModifiers.order = [
```

Comment:
How did you come up with this order? I think we usually have the visibility modifiers before things like final and override, so is there a reason not to keep it that way?

Reply:
I copied from somewhere, maybe the intellij settings? We can just change it back

Reply:
Actually, override might more often be already in the front, so maybe that's something to keep. Should check later.

Reply:
Anyway, we should come up with one reasonable order, then apply this everywhere and stay with it. It doesn't matter too much, in any case it's better to be consistent, which currently it doesn't seem to be.

Reply:
https://docs.scala-lang.org/style/declarations.html has a preferred order:
`override`, `private`, `protected`, `implicit`, `final`, `lazy` (the last one is down below, "Lazy vals should use the lazy keyword directly before the val"

This misses `abstract`, but I guess that could be seen as kind of an opposite to `override` and thus should be in the same spot/before that? Alternatively, placing it with `final` could be an option, both can't be set together anyway.

This also misses `sealed`, as that's a modifier only for classes. This, too, could be seen as similar to either `override` or `final`. Closer to `final`, I guess?

I'm not sure I'm happy with placing `implicit` that late, but a short survey showed that others also don't think it's too important, and if there's already an official stance here, maybe we should just go with it.

Any opinions?

Reply:
Okay, since `abstract override` is a thing, `abstract` should be before `override`, and so far we have used `sealed` before `abstract` which seems to be what others do as well, so I'm inclined to put these two in the front. Still not super happy with `implicit` and `final` so late, but a standard is a standard, so let's comply with it.